File: test/codeBase-js/fsm.color/tmp/Blue.js
// version: V1

const util = require("util")
const Blue = function (context) {
	this.context = context
}

// Partial evaluate
require("./Blue.evaluate")(Blue)

// ----------
// State Logic Implement Below
// ----------

Blue.prototype.OnUpdate = function(stateData, dt) {	
	// Logic Code Here

	return this._evaluate(stateData)
};

Blue.prototype.OnEnter = function(stateData) {
	// Logic Code Here
	console.log("Blue - Enter")
};

Blue.prototype.OnExit = function(stateData) {
	// Logic Code Here
	console.log("Blue - Exit")
};

Blue.prototype.OnEvent = function(stateData, args) {
	// Logic Code Here
	console.log("Blue - On Event ", args)
};

module.exports = Blue

File: test/codeBase-cs/fsm.color/tmp/Logic/Blue.cs
using System;
using FsmCodeBase;

// ----------
// State Logic Implement Below
// ----------

namespace Fsm.Generated.FsmColor
{
    partial class Blue : BaseState
    {
        public override string GetName()
        {
            return "Blue";
        }

        public override void OnEnter(StateMachine context)
        {
            // Logic Code Here
            base.OnEnter(context);
        }

        public override string OnUpdate(float dt, StateMachine context)
        {
            // Logic Code Here
            return base.OnUpdate(dt, context);
        }

        public override void OnEvent(string name, object args, StateMachine context)
        {
            // Logic Code Here
            base.OnEvent(name, args, context);
        }

        public override void OnExit(StateMachine context)
        {
            // Logic Code Here
            base.OnExit(context);
        }
    }
}
